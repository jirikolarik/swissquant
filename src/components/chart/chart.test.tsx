import { shallow } from 'enzyme'
import React from 'react'
import { Line } from 'recharts'
import Chart, { SimpleLineChart } from '../chart'

const dummyData = [
  {
    instrumentId: 70,
    timeSeries: {
      entries: [
        {
          d: "2010-01-01",
          n: 30,
          v: 6545.91
        },
        {
          d: "2010-01-04",
          n: 12,
          v: 6631.44
        },
      ]
    }
  },   {
    instrumentId: 73,
    timeSeries: {
      entries: [
        {
          d: "2010-01-01",
          n: 50,
          v: 645.91
        },
        {
          d: "2010-01-04",
          n: 42,
          v: 631.44
        },
      ]
    }
  }
]

describe('Chart', () => {
  it('snapshot', () => {
    const wrapper = shallow(<Chart data={dummyData}  />);
    expect(wrapper).toMatchSnapshot();
  });

  it('is empty when no data are provided', () => {
    const wrapper = shallow(<Chart data={[]}  />);

    expect(wrapper.text()).toBeFalsy();
  });

  it('is empty when no data are provided', () => {
    const wrapper = shallow(<Chart data={dummyData} />);

    expect(wrapper.find(SimpleLineChart)).toBeTruthy();
  });
});

describe('SimpleLineChart', () => {
  it('is renders two lines', () => {
    const wrapper = shallow(<SimpleLineChart data={dummyData} dataKeys={[70, 73]} />);

    expect(wrapper.find(Line).length).toBe(2);
  });
});
