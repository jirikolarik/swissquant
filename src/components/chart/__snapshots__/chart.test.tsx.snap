// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <default_1
    data={
        Array [
            Object {
              "instrumentId": 70,
              "timeSeries": Object {
                "entries": Array [
                  Object {
                    "d": "2010-01-01",
                    "n": 30,
                    "v": 6545.91,
                  },
                  Object {
                    "d": "2010-01-04",
                    "n": 12,
                    "v": 6631.44,
                  },
                ],
              },
            },
            Object {
              "instrumentId": 73,
              "timeSeries": Object {
                "entries": Array [
                  Object {
                    "d": "2010-01-01",
                    "n": 50,
                    "v": 645.91,
                  },
                  Object {
                    "d": "2010-01-04",
                    "n": 42,
                    "v": 631.44,
                  },
                ],
              },
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "data": Array [
        Object {
          "70": 30,
          "73": 50,
          "name": "2010-01-01",
        },
        Object {
          "70": 12,
          "73": 42,
          "name": "2010-01-04",
        },
      ],
      "dataKeys": Array [
        70,
        73,
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "data": Array [
          Object {
            "70": 30,
            "73": 50,
            "name": "2010-01-01",
          },
          Object {
            "70": 12,
            "73": 42,
            "name": "2010-01-04",
          },
        ],
        "dataKeys": Array [
          70,
          73,
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
